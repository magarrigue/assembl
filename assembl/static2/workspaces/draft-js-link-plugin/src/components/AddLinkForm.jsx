// @flow
import React from 'react';
import { Translate } from 'react-redux-i18n';
import { Button, Checkbox, Col, ControlLabel, Form as BootstrapForm, FormControl, FormGroup } from 'react-bootstrap';
import { Form, Field } from 'react-final-form';

const FormControlAdapter = ({ input: { name, onChange, value, ...otherListeners }, ...rest }) => (
  <FormControl {...otherListeners} {...rest} onChange={event => onChange(event.target.value)} value={value || ''} />
);

const CheckboxAdapter = ({ children, input }) => <Checkbox {...input}>{children}</Checkbox>;

type Props = {
  defaultText: string,
  onSubmit: Function
};

const AddLinkForm = ({ defaultText, onSubmit }: Props) => (
  <React.Fragment>
    <Form
      initialValues={{
        text: defaultText
      }}
      onSubmit={onSubmit}
      render={({ handleSubmit }) => (
        <BootstrapForm horizontal onSubmit={handleSubmit}>
          <FormGroup controlId="url">
            <Col componentClass={ControlLabel} sm={2}>
              <Translate value="common.editor.linkPlugin.url" />
            </Col>
            <Col sm={10}>
              <Field name="url" component={FormControlAdapter} />
            </Col>
          </FormGroup>
          <FormGroup controlId="text">
            <Col componentClass={ControlLabel} sm={2}>
              <Translate value="common.editor.linkPlugin.text" />
            </Col>
            <Col sm={10}>
              <Field name="text" component={FormControlAdapter} />
            </Col>
          </FormGroup>
          <FormGroup controlId="openInNewTab">
            <Col smOffset={2} sm={10}>
              <Field name="openInNewTab" component={CheckboxAdapter} type="checkbox">
                <Translate value="common.editor.linkPlugin.openInNewTab" />
              </Field>
            </Col>
          </FormGroup>
          <FormGroup>
            <Col smOffset={2} sm={10}>
              <Button type="submit">
                <Translate value="common.editor.linkPlugin.submit" />
              </Button>
            </Col>
          </FormGroup>
        </BootstrapForm>
      )}
    />
  </React.Fragment>
);

export default AddLinkForm;