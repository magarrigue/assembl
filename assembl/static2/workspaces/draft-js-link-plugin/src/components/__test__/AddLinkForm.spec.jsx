import React from 'react';
import renderer from 'react-test-renderer';

import AddLinkForm from '../AddLinkForm';

describe('AddLinkForm component', () => {
  it('should render a form to add a link', () => {
    const onSubmitSpy = jest.fn();
    const props = {
      onSubmit: onSubmitSpy
    };
    const component = renderer.create(<AddLinkForm {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});