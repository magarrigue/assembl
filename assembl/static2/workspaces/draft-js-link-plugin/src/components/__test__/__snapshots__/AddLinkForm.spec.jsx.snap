// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddLinkForm component should render a form to add a link 1`] = `
<form
  className="form-horizontal"
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      className="col-sm-2 control-label"
      htmlFor="url"
    >
      <span
        className={undefined}
        style={undefined}
      >
        URL
      </span>
    </label>
    <div
      className="col-sm-10"
    >
      <input
        className="form-control"
        id="url"
        meta={
          Object {
            "active": false,
            "data": Object {},
            "dirty": false,
            "dirtySinceLastSubmit": false,
            "error": undefined,
            "initial": undefined,
            "invalid": false,
            "pristine": true,
            "submitError": undefined,
            "submitFailed": false,
            "submitSucceeded": false,
            "touched": false,
            "valid": true,
            "visited": false,
          }
        }
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        render={undefined}
        type={undefined}
        value=""
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      className="col-sm-2 control-label"
      htmlFor="text"
    >
      <span
        className={undefined}
        style={undefined}
      >
        Link text
      </span>
    </label>
    <div
      className="col-sm-10"
    >
      <input
        className="form-control"
        id="text"
        meta={
          Object {
            "active": false,
            "data": Object {},
            "dirty": false,
            "dirtySinceLastSubmit": false,
            "error": undefined,
            "initial": undefined,
            "invalid": false,
            "pristine": true,
            "submitError": undefined,
            "submitFailed": false,
            "submitSucceeded": false,
            "touched": false,
            "valid": true,
            "visited": false,
          }
        }
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        render={undefined}
        type={undefined}
        value=""
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <div
      className="col-sm-10 col-sm-offset-2"
    >
      <div
        className="checkbox"
        style={undefined}
      >
        <label
          title=""
        >
          <input
            checked={false}
            disabled={false}
            name="openInNewTab"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <span
            className={undefined}
            style={undefined}
          >
            Open link in new tab
          </span>
        </label>
      </div>
    </div>
  </div>
  <div
    className="form-group"
  >
    <div
      className="col-sm-10 col-sm-offset-2"
    >
      <button
        className="btn btn-default"
        disabled={false}
        type="submit"
      >
        <span
          className={undefined}
          style={undefined}
        >
          Validate
        </span>
      </button>
    </div>
  </div>
</form>
`;
